### DISCUSSION TITLE
Regarding Dependency Folder

### DISCUSSION THREAD

**Discussion by Riturajvats** [apache/airflow#24777] [Q&A]
I have added the dependency folder in this current image in order to install the additional dependency  in the image.

ADD ./dependencies/postfix_restart.sh  /opt

This is how we are adding the dependency in the image but we are getting error during the build.

ERROR: failed to compute cache key: /dependencies/postfix_restart.sh  not found: not found

**Comment by potiuk** [comment]
Depends how you run your image - you have to check if the files you add are in context. Check .dockerignore (you forgot to describe in detail how you are running your build so I need to guess).

**Reply by Riturajvats** [reply]
Thanks @potiuk , By adding folder to .dockerignore file resolve the issue

**Reply by potiuk** [reply]
If that's the case, you have two options. 

1) If you want to continue all airflow sources, the preffered way for you will be to add the "dependencies" folder to "docker-context-files" . This folder is speciflcally foreseen for things like that - any files placed there are added to the context because it is already present in .dockerignore

This is described in detail here: https://airflow.apache.org/docs/docker-stack/build.html#using-docker-context-files including examples.

2) As of airlfow 2.3.* you do not have to do it. As of Airflow 2.3, the Dockerfile of Airflow is "standalone" - i.e. it does not need any other sources of Airlfow. See the information in the changelog https://airflow.apache.org/docs/docker-stack/changelog.html#airflow-2-3 - you can just copy Airlfow's Dockerfile (and no .dockerignore) and it will continue to work (you can continue passing the build arguments to cusomize your image as you want: https://airflow.apache.org/docs/docker-stack/build.html#customizing-the-image
